### DISCUSSION TITLE
Are aria roles now redundant in react table v7

### DISCUSSION THREAD

**Discussion by scmccoy** [TanStack/table#2723] [General] [answered]
A discussion recently came up while using react table v7.  (edited)
There was a shift from using div's to allowing semantic html - great for accessibility and readability. v7 still adds aria-roles to these elements though. 
```<table role="table"></table>```
Per [W3 guidelines](https://w3c.github.io/html-aria/#rules-wd)

> It is NOT RECOMMENDED for web developers to set the ARIA role and aria-* attributes to values that match the implicit ARIA semantics defined in the [table](https://w3c.github.io/html-aria/#docconformance). Doing so is unnecessary and can potentially lead to unintended consequences.

Elements ```table```, ```tr```, ```th``` etc are included in the [table](https://w3c.github.io/html-aria/#docconformance) referenced in above quote. 

I have not experienced any unintended consequences and have not found any reports about issues in relation to this. 
Our preference is to use semantic html but not have to strip out the ```roles``` if they cause issues for users. 
Is there a reason ```roles``` have remained with semantic html?
If there is a reason I would like to hear about it and if not, we can maybe make the redundant ```roles``` optional.
Best,
Shane

**Comment by tannerlinsley** [comment, marked as answer]
Not all tables are built with table elements. Eg div tables.

Tanner Linsley

**Reply by racheljianijiang** [reply]
hey Tanner,

the issue of redundant aria roles is causing our website to fail the accessibility scan via Siteimprove. Can you please clarify more about how to solve this issue? or do we have option to remove the redundant roles? Thanks in advance!

Rachel

**Reply by ben8jam** [reply]
I'm in same boat. Site Improve fails with redundant roles. is there any way to remove them from being rendered when the field is already a html table element?

Well to get around this - I simply removed all the references to {...getXYZ props} from each of the table elements. The only one in there was row="xxx"
